atoMEC/mathtools.py

```diff
@@ -471,3 +471,25 @@ def lorentzian(x, x0, gamma):
     """
     lorentzian_ = (gamma / np.pi) * (1.0 / (gamma ** 2 + (x - x0) ** 2))
     return lorentzian_
+
+
+def grad_den(den, rgrid, xgrid):
```

Comment:
This would be better written as:

`def grad_func(func, xgrid):
    ...
`

Since it is valid for a general functional (not just the density), and it would be easier to take as input just the `xgrid`.